hail_search/queries/mito.py

```diff
@@ -181,10 +181,28 @@ def _prefilter_entries_table(self, ht, parsed_intervals=None, exclude_intervals=
             ht = hl.filter_intervals(ht, parsed_intervals, keep=False)
         return ht
 
-    def _get_consequence_filter(self, allowed_consequence_ids, annotation_exprs):
+    def _get_consequence_filter(self, allowed_consequence_ids, allowed_consequences, annotation_exprs):
+        canonical_consequences = {
+            c.replace('__canonical', '') for c in allowed_consequences if c.endswith('__canonical')
+        }
+        canonical_consequences -= allowed_consequences
```

Comment:
Is this line needed if we have the `c.endswith('__canonical')` in the set expression above?

Reply:
yeah because if a user selected both the canonical filter and the regular filter for that same consequence, by definition all the canonicals will also pass the regular so theres no need to do any special canonical handling for that consequence